rholang/src/main/scala/coop/rchain/rholang/interpreter/util/AddressTools.scala

```diff
@@ -0,0 +1,65 @@
+package coop.rchain.rholang.interpreter.util
+
+import java.nio.charset.StandardCharsets
+
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.rholang.interpreter.util.codec.Base58
+
+final case class Address(val prefix: Array[Byte], val keyHash: Array[Byte])
+
+class AddressTools(prefix: Array[Byte], keyLength: Int, checksumLength: Int) {
+  private def computeChecksum(toCheck: Array[Byte]): Array[Byte] =
+    Blake2b256.hash(toCheck).take(checksumLength)
+
+  /**
+    * Creates an Address given a public key.
+    *
+    * @param pk the public key from which the address is derived
+    * @return None if the key length is invalid or Some if the address was created successfully
+    */
+  def fromPublicKey(pk: Array[Byte]): Option[String] =
```

Comment:
If we use the existing `PublicKey` type for the input parameter, we get type-level enforcements for what can be used to construct addresses.